avocado/utils/asset.py

```diff
@@ -205,7 +194,12 @@ def _verify(self):
         else:
             return False
 
-    def _get_local_file(self, path):
+    def _get_local_file(self, url_obj):
+        if os.path.isdir(url_obj.path):
```

Comment:
yes, this is a better place for per-implementation details and those functions are internal, therefor we can modify them freely.